app/services/reports/querier.rb

```diff
@@ -2,38 +2,53 @@ module Reports
 
   class Querier
 
+    # This seems to be ideal for NU/Oracle, but we can consider tweaking for MySQL
+    BATCH_SIZE = 500
+
     attr_reader :order_status_id, :current_facility, :date_range_field,
-                :date_range_start, :date_range_end, :extra_includes
+                :date_range_start, :date_range_end, :extra_includes,
+                :extra_joins, :extra_preloads, :batch_size, :options
 
     def initialize(options = {})
+      @options = options
       @order_status_id = options[:order_status_id]
       @current_facility = options[:current_facility]
       @date_range_field = options[:date_range_field]
       @date_range_start = options[:date_range_start]
       @date_range_end = options[:date_range_end]
-      @extra_includes = options[:includes]
       @transformer_options = options[:transformer_options]
+      @batch_size = options[:batch_size] || BATCH_SIZE
     end
 
     def perform
       OrderDetailListTransformerFactory.instance(order_details).perform(@transformer_options)
     end
 
     def order_details
-      return [] if order_status_id.blank?
-
-      includes = default_includes
-      includes.concat(extra_includes) if extra_includes.present?
+      return OrderDetail.none if order_status_id.blank?
 
       OrderDetail.where(order_status_id: order_status_id)
                  .for_facility(current_facility)
                  .action_in_date_range(date_range_field, date_range_start, date_range_end)
+                 .joins(*joins)
                  .includes(*includes)
+                 .preload(*preloads)
                  .merge(Order.purchased)
+                 .find_each(batch_size: batch_size)
+    end
+
+    private
+
+    def joins
+      [:order, :account] + Array(options[:joins])
```

Comment:
I don't object to us having `Array(options[:joins])` here, and similar below, but I'm wondering what the expected use case is.

Reply:
Wrapping with `Array()` allows `options[:joins]` to be `nil`, a single item, or an array. The big advantage is handling the `nil` case, which will be converted to an empty array so the `+` will still work.

Reply:
Oh, I meant when are we expecting to be passing in options[:joins] that is not nil, essentially?

Reply:
I'm not 100%. I had `reservation` being passed in there, but turns out that means the query would exclude non-reservation orders. I do think I like having the option there so the three methods are consistent.